more_itertools/recipes.py

```diff
@@ -287,7 +294,7 @@ def random_product(*args, **kwds):
         ('b', '2', 'c', '2')
 
     """
-    pools = map(tuple, args) * kwds.get('repeat', 1)
+    pools = [tuple(pool) for pool in args] * kwds.get('repeat', 1)
```

Comment:
Yeah - Py3 `map` doesn't return a multiply-able `list` like Py2's does. Could wrap the `map(tuple ,args)` in a `list()`, but the comprehension is cleaner.